Taylor and Francis Group.js

```diff
@@ -0,0 +1,287 @@
+	"creator": "Abe Jellinek",
+	"target": "^https?://(www\\.)?taylorfrancis\\.com/books|chapters|search",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2021-05-21 16:46:25"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2021 Abe Jellinek
+	
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+const FIELDS = {
+	Edition: 'edition',
+	'First Published': 'date',
+	'eBook Published': 'date',
+	Imprint: 'publisher',
+	Pages: 'numPages',
+	DOI: 'extra',
+	'eBook ISBN': 'ISBN',
+	'Pub. Location': 'place'
+};
+
+function detectWeb(doc, url) {
+	if (url.includes('/chapters/')) {
+		return "bookSection";
+	}
+	if (url.includes('/books/')) {
+		return "book";
+	}
+	if (getSearchResults(doc, true)) {
+		return "multiple";
+	}
+	return false;
+}
+
+function getSearchResults(doc, checkOnly) {
+	var items = {};
+	var found = false;
+	var rows = doc.querySelectorAll('search-results-product > a');
+	for (let row of rows) {
+		let href = row.href;
+		let title = ZU.trimInternal(text(row, 'h1'));
+		if (!href || !title) continue;
+		if (checkOnly) return true;
+		found = true;
+		items[href] = title;
+	}
+	return found ? items : false;
+}
+
+function doWeb(doc, url) {
+	let itemType = detectWeb(doc, url);
+	if (itemType == "multiple") {
+		Zotero.selectItems(getSearchResults(doc, false), function (items) {
+			if (items) ZU.processDocuments(Object.keys(items), scrape);
+		});
+	}
+	else {
+		scrape(doc, url, itemType);
+	}
+}
+
+function scrape(doc, url, itemType) {
+	if (!itemType) {
+		itemType = detectWeb(doc, url);
+	}
+
+	let item = new Zotero.Item(itemType);
+	item.title = text(doc, 'h1.product-banner-title');
+	item.libraryCatalog = '';
+	item.extra = '';
+	let subtitle = text(doc, '.product-banner-subtitle');
+	if (subtitle) {
+		item.title += ": " + subtitle;
+	}
+	if (itemType == 'bookSection') {
+		item.bookTitle = text(doc, 'a.product-book-link');
+	}
+	
+	for (let creator of text(doc, '.product-banner-author-name').split(", ")) {
+		let creatorType = text(doc, '.product-banner-author').includes('Edited')
+			? 'editor'
+			: 'author';
+		item.creators.push(ZU.cleanAuthor(creator, creatorType, false));
+	}
+	
+	for (let row of doc.querySelectorAll('product-more-details .display-row')) {
```

Comment:
So they're just…making up tag names? Seems like there are classes, too. Not sure which is more likely to remain stable.

Reply:
Angular is [like that](https://angular.io/guide/elements). I think the tag names are more likely to remain stable, just because they describe the semantics rather than style.